### PR TITLE
update 4.3.1

### PR DESCRIPTION
**Pull request by Arishamays1**
## ☆ Ydata-profiling Update 4.3.1 (PKG-2133)☆
[Jira Ticket](https://anaconda.atlassian.net/browse/PKG-2133)
[Upstream](https://github.com/ydataai/ydata-profiling/tree/v4.3.1)
[Dependencies ](https://github.com/ydataai/ydata-profiling/blob/v4.3.1/requirements.txt)
# Changes
- Updated version number and `sha256`
- Updated pinnings and dependencies

### CODE REVIEW

File: recipe/meta.yaml [hunk 1]
```diff
@@ -1,19 +1,19 @@
 {% set name = "ydata-profiling" %}
-{% set version = "4.1.1" %}
+{% set version = "4.2.0" %}
```
Comment (skupr-anaconda) [inline recipe/meta.yaml]: Could you please update to 4.3.1 https://github.com/ydataai/ydata-profiling/releases/tag/v4.3.1 ?

Comment (Arishamays1) [reply recipe/meta.yaml]: sure

File: recipe/meta.yaml [hunk 2]
```diff
@@ -38,21 +38,22 @@ requirements:
...
+    - multimethod >=1.4,<2
     # metrics
-    - statsmodels >=0.13.2,<0.14
-    - typeguard >=2.13.2,<2.14
+    - statsmodels >=0.13.2,<1
+    # type checking
+    - typeguard >=2.13.2,<3
     - imagehash ==4.3.1
-    - pillow
+    - wordcloud >=1.8.2.2
```
Comment (skupr-anaconda) [inline recipe/meta.yaml]:
```suggestion
    - wordcloud >=1.9.1
```
https://github.com/ydataai/ydata-profiling/blob/116fb8257ff49b0071ac321779739643650fe59e/requirements.txt#L25C1-L25C17

File: recipe/meta.yaml [hunk 3]
```diff
@@ -38,21 +38,22 @@ requirements:
...
+    - dacite >=1.6
```
Comment (skupr-anaconda) [inline recipe/meta.yaml]:
```suggestion
    - dacite >=1.8
```
https://github.com/ydataai/ydata-profiling/blob/116fb8257ff49b0071ac321779739643650fe59e/requirements.txt#L26C1-L26C12

File: recipe/meta.yaml [hunk 4]
```diff
@@ -38,21 +38,22 @@ requirements:
...
   run_constrained:
     - jupyter-client >=5.3.4
     - jupyter-core >=4.6.3
     - tangled-up-in-unicode 0.2.0
-    - ipywidgets >=7.5.1
```
Comment (skupr-anaconda) [inline recipe/meta.yaml]: `ipywidgets >=7.5.1` is still required here https://github.com/ydataai/ydata-profiling/blob/116fb8257ff49b0071ac321779739643650fe59e/setup.py#L38

**Comment by JeanChristopheMorinPerso** [comment]
@Arishamays1 can you fix the links in the PR description please? They point to `python-pyproject-metadata` instead of `ydata-profiling`.

File: recipe/meta.yaml [hunk 5]
```diff
@@ -38,21 +38,22 @@ requirements:
...
-    - multimethod >=1.4,<1.10
+    - multimethod >=1.4,<2
     # metrics
-    - statsmodels >=0.13.2,<0.14
-    - typeguard >=2.13.2,<2.14
+    - statsmodels >=0.13.2,<1
+    # type checking
+    - typeguard >=2.13.2,<3
     - imagehash ==4.3.1
-    - pillow
```
Comment (JeanChristopheMorinPerso) [inline recipe/meta.yaml:50]: Why is pillow removed? Upstream are using `visions[type_image_path]` which pulls in `imagehash` and `pillow` and looking at the code, both seem to be used explicitly in the code.
* Pillow usage: https://github.com/search?q=repo%3Adylan-profiler%2Fvisions+PIL&type=code
* imagehash: https://github.com/dylan-profiler/visions/blob/dca86aa1f6be9d5872f7cce91414e1fa33bd1b85/src/visions/utils/images/image_utils.py#L5

**Review by JeanChristopheMorinPerso — CHANGES_REQUESTED** [review]

File: recipe/meta.yaml (hunk 2 above)
Comment (Arishamays1) [reply recipe/meta.yaml]: Got it thanks!

File: recipe/meta.yaml (hunk 3 above)
Comment (Arishamays1) [reply recipe/meta.yaml]: Made the amendment! Thanks!

File: recipe/meta.yaml (hunk 5 above)
Comment (Arishamays1) [reply recipe/meta.yaml:50]: Thanks for catching this! I did not see it in the required dependencies in `setup.py`, so I was unsure if it was still required. I'll add it back.

**Review by skupr-anaconda — APPROVED** [review]

File: recipe/meta.yaml (hunk 4 above)
Comment (JeanChristopheMorinPerso) [reply recipe/meta.yaml]: @Arishamays1 can you add this back please?

Comment (Arishamays1) [reply recipe/meta.yaml]: Its been added!

Comment (JeanChristopheMorinPerso) [reply recipe/meta.yaml]: You added it in the runtime dependencies. It should be in the `run_constrained` block.

**Comment by Arishamays1** [comment]
> @Arishamays1 can you fix the links in the PR description please? They point to `python-pyproject-metadata` instead of `ydata-profiling`.

Fixed

**Review by JeanChristopheMorinPerso — APPROVED** [review]

File: recipe/meta.yaml (hunk 4 above)
Comment (Arishamays1) [reply recipe/meta.yaml]: Ahh thanks for letting me know. Updated

**Review by varlackc — APPROVED** [review]
LGTM

**Pull request merged**